pydantic/types.py

```diff
@@ -505,10 +508,28 @@ def list_length_validator(cls, v: 'Optional[List[T]]') -> 'Optional[List[T]]':
 
         return v
 
+    @classmethod
+    def unique_items_validator(cls, v: 'Optional[List[T]]') -> 'Optional[List[T]]':
+        try:
+            if v and len(set(v)) != len(v):
+                raise errors.ListUniqueItemsError(not_unique=len(v) - len(set(v)))
+        except TypeError:
+            # failover for unhashable types
+            unique = list()
 
-def conlist(item_type: Type[T], *, min_items: int = None, max_items: int = None) -> Type[List[T]]:
+            if v and len([unique.append(i) for i in v if i not in unique]) != len(v):  # type: ignore
```

Comment:
modifying unique inside a comprehension is somewhat unorthodox, also (if we don't care about how many items are not unique, which I will mostly be the case) we can do this faster by raising the error as soon as we find a duplicate.

Something like
```py
for i, value in enumerate(v, start=1):
    if value in v[i:]:
        raise errors.ListUniqueItemsError()
```

I haven't done any profiling but I feel something like this should be quicker.

Reply:
I've just profiled it and you're totally right about performance. Also, the difference between the `set` and the loop checks is almost negligible (even when the duplicate is at the end of the iterable).

I also agree that the most of the time we don't care about how many items aren't unique, and furthermore, this calculation reduces the performance. So I'm going to refactor this PR to push it again.